### PR TITLE
modified hardcoded links in docs

### PR DESCRIPTION
**Pull request by CharityKithaka**
[Modified hardcoded links to follow the rSt way of writing.]

## Changes
Changed all links in all the docs. 

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by CharityKithaka** [comment]
Hey @skrawcz the changes.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "esbonio.sphinx.confDir": ""
+}
```
Comment (skrawcz) [inline .vscode/settings.json:3]: please remove this file.

File: docs/extensions.rst [hunk 2]
```diff
@@ -90,8 +90,7 @@ Single Machine:
 `Dask <https://dask.org/>`_ is a very easy way to enable multi-processing on a single machine. This enables you to
 easily make use of multiple CPU cores.
 
-What this doesn't help with is data scale, as you're still limited to what fits in memory on your machine.
-
+What this doesn't help with is data scale, as you're still limited to what fits in memory on your mach
```
Comment (skrawcz) [inline docs/extensions.rst:93]: revert this change please.

File: docs/hamilton-community.rst [hunk 3]
```diff
@@ -6,8 +6,7 @@ If you want to bounce ideas, ask questions, or just see what the small community
 
 We have a bunch of channels on Slack that are designed to help you get the most out of Hamilton!
 
-Slack space name: `https://hamilton-opensource.slack.com/ <https://hamilton-opensource.slack.com/>`_
-
-Slack invite link: `https://join.slack.com/t/hamilton-opensource/shared\_invite/zt-1bjs72asx-wcUTgH7q7QX1igiQ5bbdcg <https://join.slack.com/t/hamilton-opensource/shared\_invite/zt-1bjs72asx-wcUTgH7q7QX1igiQ5bbdcg>`_
+Slack space name: `slack community <https://hamilton-opensource.slack.com/>`_
```
Comment (skrawcz) [inline docs/hamilton-community.rst:9]:
```suggestion
Slack space name: `hamilton-opensource <https://hamilton-opensource.slack.com/>`_
```

File: docs/hamilton-community.rst [hunk 4]
```diff
@@ -6,8 +6,7 @@ If you want to bounce ideas, ask questions, or just see what the small community
...
 
+Slack invite link: `slack link <https://join.slack.com/t/hamilton-opensource/shared\_invite/zt-1bjs72asx-wcUTgH7q7QX1igiQ5bbdcg/>`_
```
Comment (skrawcz) [inline docs/hamilton-community.rst:11]:
```suggestion
Slack invite link: `click me to join! <https://join.slack.com/t/hamilton-opensource/shared\_invite/zt-1bjs72asx-wcUTgH7q7QX1igiQ5bbdcg>`_
```

**Review by skrawcz — CHANGES_REQUESTED** [review]
Thanks for the touch ups! But a couple of things to fix up please.

1. Removing the vscode file.
2. There was a paragraph that I think you by accident deleted some words from.
3. The reST guide has a `trailing slash` i.e. `/` in its example. We don't actually need that trailing slash `/` for our links. Could you remove it for the links that you added it to please?
e.g. `<https://medium.com/avmconsulting-blog/running-a-python-application-on-kubernetes-aws-56609e7cd88c/>`_ should just be `<https://medium.com/avmconsulting-blog/running-a-python-application-on-kubernetes-aws-56609e7cd88c>`_

File: docs/hamilton-community.rst (hunk 3 above)
Comment (skrawcz) [reply docs/hamilton-community.rst:9]: Just suggesting some better wording here.

File: docs/hamilton-community.rst (hunk 4 above)
Comment (skrawcz) [reply docs/hamilton-community.rst:11]: Just suggesting some better wording here.

**Comment by CharityKithaka** [comment]
hey @skrawcz worked on the changes.

**Comment by skrawcz** [comment]
@CharityKithaka the pre-commit check is failing. Could you:

1. install pre-commit. `pre-commit install`
2. run it -- `pre-commit run --all-files` -- it should fix the two files that are failing.
3. commit the changes.
If you need help see https://github.com/stitchfix/hamilton/blob/main/developer_setup.md#set-up-your-local-dev-environment and https://pre-commit.com/

**Comment by CharityKithaka** [comment]
Hey @skrawcz everything is in order now.

**Review by skrawcz — APPROVED** [review]
Thanks @CharityKithaka! I will merge this now. Sorry I have been a little swamped this week; I will try to get you another task tomorrow/the day after.

**Comment by CharityKithaka** [comment]
Hey, @skrawcz that would be great. I will truly appreciate it.

**Pull request merged**